classes/ZipAction.php

```diff
@@ -165,21 +165,37 @@ public function extract(string $from_file, string $to_dir): bool
         }
 
         for ($i = 0; $i < $zip->numFiles; ++$i) {
-            if (!$zip->extractTo($to_dir, [$zip->getNameIndex($i)])) {
-                $error = $zip->getStatusString();
+            $fileName = $zip->getNameIndex($i);
+            if (!$zip->extractTo($to_dir, [$fileName])) {
+                $stat = $zip->statIndex($i);
+                $filePath = $stat['name'];
+
+                $fullSystemPath = rtrim($to_dir, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR . $filePath;
+                $md5FromZip = md5($zip->getFromIndex($i));
+                $md5FromSystem = is_file($fullSystemPath) ? md5_file($fullSystemPath) : null;
+                $isFileUnchanged = $md5FromZip === $md5FromSystem;
+
+                $error = $zip->getStatusString() ?: 'Unknown extraction error (possibly permission denied or filesystem error)';
+
+                if ($isFileUnchanged) {
+                    $error .= ' – file unchanged, skipped';
+                }
 
-                $this->logger->error(
+                $this->logger->warning(
                     $this->translator->trans(
                         'Could not extract %file% from archive: %error%',
                         [
-                            '%file%' => $zip->statIndex($i)['name'],
+                            '%file%' => $filePath,
```

Comment:
On https://github.com/PrestaShop/autoupgrade/pull/1317/files#diff-8e29e32fd69cad53f2beb198e81f9338413e6f7db022df279eaa5b2cba5b987dR218, I've set a new variable `$nameIndex` based on the value of `$zip->getNameIndex($i);`, if you want to avoid an incoming conflict.